netket/experimental/driver/tdvp.py

```diff
@@ -0,0 +1,395 @@
+
+            for step in self.iter(T, tstops=tstops):
+                log_data = self.estimate(obs)
+
+                # if the cost-function is defined then report it in the progress bar
+                if self._loss_stats is not None:
+                    pbar.set_postfix_str(self._loss_name + "=" + str(self._loss_stats))
+                    log_data[self._loss_name] = self._loss_stats
+
+                # Execute callbacks before loggers because they can append to log_data
+                for callback in callbacks:
+                    if not callback(step, log_data, self):
+                        callback_stop = True
+
+                for logger in loggers:
+                    logger(self.step_value, log_data, self.state)
+
+                if len(callbacks) > 0:
+                    if mpi.mpi_any(callback_stop):
+                        break
+
+                # Reset the timing of tqdm after the first step, to ignore compilation time
+                if first_step:
+                    first_step = False
+                    pbar.unpause()
+
+                # Update the progress bar
+                pbar.update(np.asarray(self.step_value - old_step))
+                old_step = self.step_value
+
+            # Final update so that it shows up filled.
+            pbar.update(np.asarray(self.step_value - old_step))
+
+        # flush at the end of the evolution so that final values are saved to
+        # file
+        for logger in loggers:
+            logger.flush(self.state)
+
+        return loggers
+
+    def _log_additional_data(self, obs, step):
+        obs["t"] = self.t
+
+    @property
+    def _default_step_size(self):
+        # Essentially means
+        return None
+
+    @property
+    def step_value(self):
+        return self.t
+
+    @property
+    def dt(self):
+        return self._integrator.dt
+
+    @property
+    def t(self):
+        return self._integrator.t
+
+    @t.setter
+    def t(self, t):
+        self._integrator.t = jnp.array(t, dtype=self._integrator.t)
+
+    @property
+    def t0(self):
+        """
+        The initial time set when the driver was created.
+        """
+        return self._t0
+
+    def __repr__(self):
+        return f"{type(self).__name__}(step_count={self.step_count}, t={self.t})"
+
+    def info(self, depth=0):
+        lines = [
+            "{}: {}".format(name, info(obj, depth=depth + 1))
+            for name, obj in [
+                ("generator     ", self.generator),
+                ("integrator    ", self._integrator),
+                ("linear solver ", self.linear_solver),
+                ("state         ", self.state),
+            ]
+        ]
+        return "\n{}".format(" " * 3 * (depth + 1)).join([str(self)] + lines)
+
+    def _odefun(self, t, w, **kwargs):
+        """
+        Internal method which dispatches to the actual ODE system function.
+        """
+        return dwdt_mcstate(self.state, self, t, w, **kwargs)
```

Comment:
```suggestion
        return dwdt(self.state, self, t, w, **kwargs)
```

or any better name we come up with, really.